### DISCUSSION TITLE
Is targetMethod in System.Reflection.DispatchProxy.Invoke() really nullable?

### DISCUSSION THREAD

**Discussion by yyjdelete** [dotnet/runtime#60501] [General]
Is targetMethod in System.Reflection.DispatchProxy.Invoke() really nullable?
https://github.com/dotnet/runtime/blob/main/src/libraries/System.Reflection.DispatchProxy/src/System/Reflection/DispatchProxy.cs#L25

If it's right, when will `targetMethod` be null, and what's the right way to handle it in override method?

**Comment by huoyaoyuan** [comment]
The source browser shows only one override which rejects `null`.

**Reply by huoyaoyuan** [reply]
Tagging area owner @jeffhandley for decision

**Reply by jeffhandley** [reply]
Relaying to @krwq or @buyaa-n.

**Reply by buyaa-n** [reply]
Don't remember why I annotated `targetMethod` as nullable, and could not think of scenario it could be null, probably wrong annotation any suggestions @stephentoub?

**Reply by krwq** [reply]
Quickly skimming through the code it looks like annotation issue. We'd need to trace back the value passed to that method from here: https://github.com/dotnet/runtime/blob/main/src/libraries/System.Reflection.DispatchProxy/src/System/Reflection/DispatchProxyGenerator.cs#L453 to double check. I'd recommend to file an issue about it.
